### ISSUE TITLE
TestAddressManagerNoRequestedIP failing with make test

### ISSUE THREAD

**Issue by RainbowMango** [kubernetes/kubernetes#86687]
**What happened**:
Unit test failed by command `make test`:

```
--- FAIL: TestAddressManagerNoRequestedIP (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1be5f79]

goroutine 37 [running]:
testing.tRunner.func1(0xc0004a4100)
/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x1f118c0, 0x34eb510)
/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce.(*Cloud).getRegionLink(...)
/root/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce/gce_zones.go:90
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce.fakeGCECloud(0x22e1693, 0xc, 0x22e05f8, 0xb, 0x22e2a80, 0xd, 0x22e2a8d, 0xd, 0x22e5093, 0xf, ...)
/root/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce/gce_util.go:72 +0x409
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce.TestAddressManagerNoRequestedIP(0xc0004a4100)
/root/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce/gce_address_manager_test.go:38 +0x51
testing.tRunner(0xc0004a4100, 0x23afe70)
/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
/usr/local/go/src/testing/testing.go:960 +0x350
FAILk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce0.018s
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Just run command `make test`.

**Anything else we need to know?**:
Happened on [master branch](https://github.com/kubernetes/kubernetes/tree/d605766c892bff09d182ae96d190e426acd2d7e6).

I first met this issue at v1.15.

**Environment**:
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.0-alpha.0.1670+0d58709016f335", GitCommit:"0d58709016f3358f34a52c12087d707a64cc6cb0", GitT
reeState:"clean", BuildDate:"2019-12-12T06:21:17Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
- Kernel (e.g. `uname -a`):

```
Linux ecs-d8b6 3.10.0-957.12.2.el7.x86_64 #1 SMP Tue May 14 21:24:32 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by liggitt** [comment]
I would be curious to know how https://testgrid.k8s.io/sig-release-master-blocking#bazel-test-master&include-filter-by-regex=%2F%2Fstaging%2Fsrc%2Fk8s.io%2Flegacy-cloud-providers%2Fgce%3Ago_default_test is passing then

**Comment by RainbowMango** [comment]
The reason is that `Google Application Default Credentials` is required here:
https://github.com/kubernetes/kubernetes/blob/bd239d42e463bff7694c30c994abd54e4db78700/staging/src/k8s.io/legacy-cloud-providers/gce/gce_fake.go#L65

I guess all CI machinery has a `Google Application Default Credentials`.  This will fail in all environments without a `Google Application Default Credentials`.

**Comment by liggitt** [comment]
ah, fixed in #86713

**Comment by RainbowMango** [comment]
Great!
